### PR TITLE
Add router support for running multiple augmentation loops

### PR DESCRIPTION
**Pull request by msempere**
Usage example:
```
/* -*- js-mode -*- */
[
    {
        "exchangeType": "mock",
        "listenPort": 12339,
        "bindHost": "0.0.0.0",
        "auctionVerb": "POST",
        "auctionResource": "/auctions",
        "performNameLookup": false,
        "numAugmentationLoops": 3
    }
]
```

### CODE REVIEW

File: rtbkit/core/router/router.cc [hunk 1]
```diff
@@ -1417,8 +1451,8 @@ augmentAuction(const std::shared_ptr<AugmentationInfo> & info)
             wakeupMainLoop.signal();
         };
 
-    augmentationLoop.augment(info, Date::now().plusSeconds(augmentationWindow.count()),
-                             onDoneAugmenting);
+    augmentationLoops.at(numAuctions % augmentationLoops.size())->augment(
```
Comment (oktal) [inline rtbkit/core/router/router.cc]: I would avoid using `at()` in this case, using `operator[]` is safe enough as you're clearly controlling the bound with the modulo so you don't need any additional bound checking.

Comment (oktal) [inline rtbkit/core/router/router.cc]: Also, you're using a round-robin scheme, right ?

**Comment by oktal** [comment]
Aide from a minor comment, everything looks good.

File: rtbkit/core/router/router.cc (hunk 1 above)
Comment (msempere) [inline rtbkit/core/router/router.cc]: Hi @oktal , yes, I'm following a round-robin approach. 
Regarding the at() or [] operator, yes, it's ok, I can add the new modification you are proposing.

**Pull request closed without merging**